router.go

```diff
@@ -67,51 +85,123 @@ func (m MapRouter) Register(rs []transport.Procedure) {
 			service:   r.Service,
 			procedure: r.Name,
 		}
-		m.entries[sp] = r
+
+		if r.Encoding == "" {
+			// Protect against masking encoding-specific routes.
+			if _, ok := m.serviceProcedures[sp]; ok {
+				panic(fmt.Sprintf("Cannot register multiple handlers for every encoding for service %q and procedure  %q", sp.service, sp.procedure))
```

Comment:
I strongly disagree - why would someone want to override a procedure? This will mess with a lot of logic and is effectively a bug